src/modules/iotjs_module_process.c

```diff
@@ -349,6 +349,26 @@ JS_FUNCTION(DLOpen) {
   (*initfn)(exports);
   return exports;
 }
+JS_FUNCTION(MemoryUsage) {
+  size_t rss;
+  jerry_heap_stats_t stats;
+  memset(&stats, 0, sizeof(jerry_heap_stats_t));
+  int err = uv_resident_set_memory(&rss);
+  if (err) {
+    char errStr[64];
+    sprintf(errStr, "uv_resident_set_memory error with code %d", err);
+    return JS_CREATE_ERROR(COMMON, errStr);
+  }
+  jerry_get_memory_stats(&stats);
+  jerry_value_t ret = jerry_create_object();
+  iotjs_jval_set_property_number(ret, "rss", rss);
+  iotjs_jval_set_property_number(ret, "peakHeapTotal", stats.peak_allocated_bytes);
+  iotjs_jval_set_property_number(ret, "heapTotal", stats.size);
+  iotjs_jval_set_property_number(ret, "heapUsed", stats.allocated_bytes);
+  //FIXME external memory usage is not implement yet
```

Comment:
```
// FIXME...
```

Reply:
fixed :-)

Reply:
User give GC hint to V8 by calling AdjustAmountOfExternalAllocatedMemory.
There is no external memory notation in jerry now.
ShadowNode need this for application scope memory management.

Reply:
But now, background auto killing is enough:)

Reply:
is JNativeInfoType in jerry related to external memory in v8?

Reply:
Yes, it is.
typedef void (*jerry_object_native_free_callback_t) (void *native_p);
typedef struct
{
  jerry_object_native_free_callback_t free_cb; /**< the free callback of the native pointer */
} jerry_object_native_info_t;

So jerry_object_native_info_t is external memory free callback. The memory is not counted by jerry and there is no hint to GC.

Reply:
so should we set external memory usage to 0 in jerry all the time?